### PR TITLE
feat: Exclude packages that has version set by Projen from upgrade task

### PR DESCRIPTION
**Pull request by Hi-Fi**
fixes: #1659

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Hi-Fi** [comment]
@Chriscbr How does this look related to upgrade issue?

File: src/javascript/upgrade-dependencies.ts [hunk 1]
```diff
@@ -158,61 +158,83 @@ export class UpgradeDependencies extends Component {
...
+    const ncuIncludes = this.options.include?.filter(
+      (item) => !ncuExcludes.includes(item)
+    );
 
-    // run "yarn/npm install" to update the lockfile and install any deps (such as projen)
-    task.exec(this._project.package.installAndUpdateLockfileCommand);
+    const includeLength = this.options.include?.length ?? 0;
+    const ncuIncludesLength = ncuIncludes?.length ?? 0;
+    // Don't add anything if there's only includes that are excluded by e.g. setting the version.
+    if (includeLength === 0 || (includeLength > 0 && ncuIncludesLength > 0)) {
```
Comment (Chriscbr) [inline src/javascript/upgrade-dependencies.ts:193]: Why do we create the task if `includeLength === 0`?

**Review by Chriscbr — CHANGES_REQUESTED** [review]

File: .projen/tasks.json [hunk 2]
```diff
@@ -381,19 +381,19 @@
       },
       "steps": [
         {
-          "exec": "npm-check-updates --dep dev --upgrade --target=minor --reject='markmac,projen'"
+          "exec": "npm-check-updates --dep dev --upgrade --target=minor --reject='@types/fs-extra,@types/node,@typescript-eslint/eslint-plugin,@typescript-eslint/parser,eslint,jest-junit,npm-check-updates,standard-version,glob,yaml,markmac,projen'"
```
[the remark this answers is not in the document]
Comment (Hi-Fi) [reply .projen/tasks.json:384]: It's more like opinion thing, as with current solution user still gets correct version to lockfile as this step runs the upgrade step.

`ncu` doesn't seem to be updating versions in that format, so the filtering out just fully pinned (or ones with `~` could be done if needed).

File: test/upgrade-dependencies.test.ts [hunk 3]
```diff
@@ -192,6 +194,111 @@ test("github runner can be customized", () => {
   expect(upgrade.jobs.pr["runs-on"]).toEqual("self-hosted");
 });
 
+describe("upgrade task created without projen defined versions", () => {
+  test("at NodeProject", () => {
```
[the remark this answers is not in the document]
Comment (Hi-Fi) [reply test/upgrade-dependencies.test.ts:198]: These tests were also for my TDD approach, but if snapshot tests are the main thing I can remove these.

In this case TDD with snapshots would have been quite a lot of harder and slower.

File: src/javascript/upgrade-dependencies.ts (hunk 1 above)
Comment (Hi-Fi) [reply src/javascript/upgrade-dependencies.ts:193]: Include's are exlicit includes, e.g. `projen` for `upgrade-projen` task. 

So we should create task if...
- we don't have any explicit includes
- we have explicit includes, that are not pinned versions (`ncuIncludes`).

**Review by Chriscbr — APPROVED** [review]

**Pull request merged**